### PR TITLE
Add assured flag to udp connections

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

This PR completes https://github.com/DataDog/datadog-agent/pull/5358, adding separate timeouts for udp assured vs non-assured connections and adding tests. We currently use the higher "stream" udp timeout of 120s (typically) to mark a udp connection as expired. This change will allow us to use the lower timeout for non-assured udp connections, which is typically 30s. This in turn should result in a lower memory footprint since we will expire more connections per connections check, since the assured flag is usually not set for most udp connections.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

1. Run the system-probe
2. Run `socat UDP4-LISTEN:8888,fork EXEC:cat` to run a udp echo server
3. In another console, run `socat - UDP4:localhost:8888` to run a udp client. Type in at least two lines of text and hitting enter after each line. The text for each line should echo back
4. Run `curl -s --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/connections?client_id=1  | jq '.conns[] | select(.type=="udp" and (.laddr.port==8888 or .raddr.port==8888))'`. You should see two connections
5. Wait a little over 30 seconds. Run (4.) again, you should still see two connections
6. Run (5.) again every 30 seconds or so. After about 2 minutes total, you should see an empty response from the command in (4.)

### CODE REVIEW

**Review by leeavital — COMMENTED** [review]
Will you please add some background explaining why this is necessary?

**Review by leeavital — COMMENTED** [review]
The expiry logic for conntrack is already very hard to reason about, and this change further complicates it. 

What kind of memory improvement will we get?

File: pkg/network/ebpf/c/tracer-stats.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#ifndef __TRACER_STATS_H
```
Comment (leeavital) [inline pkg/network/ebpf/c/tracer-stats.h:1]: Moving common map access to a shared H file is a good refactor, but makes this harder to follow. In the future please do this in a separate PR.

File: pkg/network/ebpf/c/tracer-stats.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    if (stats->sent_bytes == 0 && recv_bytes > 0) {
+        stats->flags |= CONN_R_INIT;
+        return;
+    }
+
+    // If a three-way "handshake" was established, we mark the connection as assured
+    if ((stats->flags&CONN_L_INIT && stats->recv_bytes > 0 && sent_bytes > 0)
+        || (stats->flags&CONN_R_INIT && stats->sent_bytes > 0 && recv_bytes > 0)) {
+        stats->flags |= CONN_ASSURED;
```
Comment (leeavital) [inline pkg/network/ebpf/c/tracer-stats.h:26]: Does CONN_ASSURED need to be its own flag? Could we check that `flags&(CONN_L_INIT|CONN_R_INIT) > 0` in userspace?

File: pkg/network/ebpf/c/tracer.h [hunk 3]
```diff
@@ -11,8 +11,16 @@ typedef struct {
     __u64 sent_bytes;
     __u64 recv_bytes;
     __u64 timestamp;
+    __u32 flags;
 } conn_stats_ts_t;
 
+// Connection flags
+typedef enum {
```
Comment (leeavital) [inline pkg/network/ebpf/c/tracer.h:18]: Please document these

**Review by leeavital — COMMENTED** [review]
Some minor comments, otherwise LGTM. cc @brycekahle for a second pair of eyes.

File: pkg/network/ebpf/c/tracer-stats.h (hunk 2 above)
Comment (hmahmood) [reply pkg/network/ebpf/c/tracer-stats.h:26]: The sent and recv bytes checks also play a role, so we need that extra flag.

File: pkg/network/ebpf/c/tracer.h (hunk 3 above)
Comment (hmahmood) [reply pkg/network/ebpf/c/tracer.h:18]: 👍

**Review by leeavital — APPROVED** [review]

**Pull request merged**